README.md

```diff
@@ -1,24 +1,66 @@
 
-## Third-Party Products/Libraries used and the licenses they are covered by
-<!--- product/library and path to the LICENSE --->
-<!--- Example: <library_name> - [![GitHub](<shield_icon_link>)](<path_to_library_LICENSE>) --->
+## Introduction
+
+The application is comprised of an AWS Lambda function written in Python that gets triggered through CloudWatch Events (security alerts), and SNS Topic (budgeting alerts).
+The Lambda function formats and routes the notifications to proper RC channel based on their type and severity: 
+
+- `aws-budget` receives budget notifications when the account spending is reaching it's budget thresholds. Only accounts with a set Budget threshold participate. 
+- `aws-security-low` receives low-severity security notifications from team accounts in AWS SEA environment. These are *informational* alerts and don't require any actions most of the times.
+- `aws-security-medium` receives medium-severity security notifications from team accounts in AWS SEA environment. These are the alerts to *monitor and potentially take remedial actions*.
+- `aws-security-high` receives high-severity security notifications from team accounts in AWS SEA environment. These are alerts must be *closely watched and triaged ASAP*.
+- `aws-security-critical` receives critical-severity security notifications from team accounts in AWS SEA environment. These are the most critical alerts and *should be remediated immediately*.
+
+
+## Prerequisites
+
+In order to build, package, and run the app locally, you will need:
+
+- a `bash`-like terminal environment; testing has primarily been done using MacOS Catalina
+- `Docker`
+- `AWS SAM(Serverless Application Model)`- to build the Lambda function (on MacOS Catalina, ```brew tap aws/tap```, ```brew install aws-sam-cli```)
+- `zip`- to compress the Lambda build artifacts into a sigle zip file
+
+
+In order to deploy to AWS, you will also need:
+
+- `terraform` 12 or newer
+- the AWS CLI (on MacOS Catalina, ```brew install awscli```)
+- access to an AWS account and mechanism to get temporary (STS) credentials
+
+You would need to set up Webhook URL's for the above RC channels and add the URL's to `IncomingWebHookUrl` in a *tfvars* file prior to deployment. The *tfvars* file should contain the following variables:
+
+- aws_region = The AWS region you want to deploy the app to, e.g. `ca-central-1`
+- IncomingWebHookUrl = `"BUDGET=<Webhook URL for aws-budget>,LOW=<Webhook URL for aws-security-low>,MEDIUM=<Webhook URL for aws-security-medium>,HIGH=<Webhook URL for aws-security-high>,CRITICAL=<Webhook URL for aws-security-critical>"`
+- LambdaEnvLogLevel = Lambda event log level, e.g. `INFO`
+- LambdaTimeout = Lambda timeout in second, e.g. `10`
+
+## Build, Packaging and Testing Locally
+
+- **Build**: run ```sam build --use-container -b ../../builds``` within the lambdas/rocketchat-notification folder
```

Comment:
As an enhancement, we should attempt to make it simpler to run/deploy by incorporating calls to `sam` and zip operation into Terraform.

Reply:
Working on it...

Reply:
I'd say if you have time, pls proceed with the enhancement.

Reply:
Yes, I'll try my best!